Packages/com.dman.l-system/Runtime/UnityObjects/TurtleOperations/TurtleStemTrunkComposite/TurtleStemTrunkCompositionOperation.cs

```diff
@@ -0,0 +1,167 @@
+
+        public override void InternalCacheOperations()
+        {
+            foreach (var meshKey in stemKeys)
+            {
+                meshKey.InteralCacheOrganTemplates();
+            }
+        }
+
+        public override void WriteIntoNativeData(NativeTurtleData nativeData, TurtleNativeDataWriter writer)
+        {
+            foreach (var stemKey in stemKeys)
+            {
+                stemKey.WriteIntoNativeData(nativeData, writer);
+            }
+        }
+    }
+
+    public struct TurtleStemPlacementOperation
+    {
+        public byte materialIndex;
+        public bool willMove;
+
+        public ushort radialResolution;
+        public ScaleSource scaleSource;
+
+        public float2 lengthRadiusBase;
+        public float2 lengthRadiusScale;
+        public bool scaleIsAdditional;
+
+        public float scalePower;
+        public void Operate(
+            ref TurtleState state,
+            int indexInString,
+            SymbolString<float> sourceString,
+            NativeList<TurtleStemInstance> targetStemInstances)
+        {
+            var pIndex = sourceString.parameters[indexInString];
+
+            var localLengthRadius = lengthRadiusBase;
+            if ((scaleSource & ScaleSource.PARAMETER) != 0 && pIndex.length > 0)
+            {
+                var scale = sourceString.parameters[pIndex, 0];
+                if (scalePower != 1)
+                {
+                    scale = Mathf.Pow(scale, 1f / scalePower);
+                }
+                localLengthRadius *= (scaleIsAdditional ? new float2(1, 1) : float2.zero) + scale * lengthRadiusScale;
+            }
+            if ((scaleSource & ScaleSource.THICKNESS) != 0)
+            {
+                localLengthRadius *= new float2(1, state.thickness);
+            }
+
+            var newStemEntry = new TurtleStemInstance
+            {
+                materialIndex = materialIndex,
+                radialResolution = radialResolution,
+                orientation = state.transformation * Matrix4x4.Scale(new Vector3(localLengthRadius.x, localLengthRadius.y, localLengthRadius.y)),
+                parentIndex = state.indexInStemTree,
+                depth = state.stemDepth,
+                organIdentity = state.organIdentity,
+                extraData = state.customData
+            };
+            state.indexInStemTree = targetStemInstances.Length;
+            //if(state.indexInStemTree >= 0)
+            //{ TODO: convert the uv length calculation in this job. or write a new pre-process job to compute and propigate.
+            //    var parentPoint = targetStemInstances[state.indexInStemTree];
+            //    var distance = (parentPoint.orientation.MultiplyPoint3x4(float3.zero) - state.transformation.MultiplyPoint3x4(float3.zero)).magnitude;
+            //    state.stemDepth += distance;
+            //}
+            state.stemDepth++;
```

Comment:
add note about how this doesn't exactly work as it should. will need a post-process job to propagate stem depth.

Reply:
actually, should just fix this.